pkg/rpc/server/server.go

```diff
@@ -355,25 +377,36 @@ drainloop:
 	}
 }
 
-func (s *Server) handleWsReads(ws *websocket.Conn, resChan chan<- response.Abstract, subscr *subscriber) {
+func (s *Server) handleWsReads(ws *websocket.Conn, resChan chan<- interface{}, subscr *subscriber) {
 	ws.SetReadLimit(wsReadLimit)
 	ws.SetReadDeadline(time.Now().Add(wsPongLimit))
 	ws.SetPongHandler(func(string) error { ws.SetReadDeadline(time.Now().Add(wsPongLimit)); return nil })
 requestloop:
 	for {
-		req := new(request.In)
-		err := ws.ReadJSON(req)
+		in := request.NewIn()
+		err := ws.ReadJSON(in)
```

Comment:
Yeah, for this one either `ReadMessage` is needed or some structure unifying single request and Batch. You can make `Batch` decode a single request though.

Reply:
This implies adding some flag aimed to mark whether it is a (single request)  or a (batch-request consisting of one request). We have to distinguish these two cases because the answers should be different for them:
# Single request:
```
--> {"jsonrpc": "2.0", "method": "subtract", "params": {"subtrahend": 23, "minuend": 42}, "id": 3}
<-- {"jsonrpc": "2.0", "result": 19, "id": 3}
```

# Batch request:
```
--> [{"jsonrpc": "2.0", "method": "subtract", "params": {"subtrahend": 23, "minuend": 42}, "id": 3}]
<-- [{"jsonrpc": "2.0", "result": 19, "id": 3}]
```

That's why I unified single-request and batch-request into `In`.